js/components/menus.tsx

```diff
@@ -0,0 +1,111 @@
+import { paths } from "../paths";
+import { className } from "../util/dom";
+import { getMetaContent } from "../util/metadata";
+import { ReactElement } from "react";
+import { Link } from "react-router-dom";
+
+const MenuLink = ({
+  to,
+  icon,
+  iconExtraClassName,
+  reload,
+  newTab,
+  children,
+}: {
+  to: string;
+  icon?: string;
```

Comment:
**suggestion:** rather than an arbitrary string, could you define a custom `IconPath` type (or something like that) that only allows specifying valid icons?

Reply:
Even if it's just something simple manually defined like `type IconName = "foo" | "bar" | ...`.

Reply:
👍 60571af